src/plugins/intel_cpu/src/graph.cpp

```diff
@@ -1685,21 +1694,14 @@ bool Graph::InsertNode(NodePtr parent, NodePtr child, NodePtr node, int parentPo
     return true;
 }
 
-// Set all non const data paths precision to BF16
+// Apply inference precision configuration
 void Graph::EnforceInferencePrecision() {
     CPU_DEBUG_CAP_ENABLE(static EnforceInferPrcDebug inferPrecDebug);
-    auto inferPrec = InferenceEngine::Precision::FP32;
-    switch (getConfig().inferencePrecision) {
-    case ov::element::bf16:
-        inferPrec = InferenceEngine::Precision::BF16;
-        break;
-    case ov::element::f16:
-        inferPrec = InferenceEngine::Precision::FP16;
-        break;
-    default:
-        return;
-        break;
-    }
+
+    const auto inferPrec = convertPrecision(getConfig().inferencePrecision);
+
+    if (inferPrec == Precision::FP32)
```

Comment:
Maybe better to check allowed precisions: {BF16, FP16}?

Reply:
My idea was actually to get rid of this check after the release
And replace the checks
`if (node->getOriginalInputPrecisionAtPort(i) != Precision::FP32)`
with
`if (!node->getOriginalInputPrecisionAtPort(i).is_float())`
So it would be possible to force FP32 precision for BF16 and FP16 models, which is impossible at the moment
What do you think?